cupy_backends/hip/cupy_hipsolver.h

```diff
@@ -17,9 +17,8 @@ hipsolverStatus_t cusolverGetProperty(hipLibraryPropertyType_t type, int* val) {
     return HIPSOLVER_STATUS_SUCCESS;
 }
 
-typedef enum hipsolverDnParams_t {};
```

Comment:
@shbiswas834  @AdrianAbeyta  Is there a reason we deleted this logic? If you see this is not useful anymore in our latest releases, keep a conditon that it needed below a specific version like this.

```
#if HIP_VERSION less than <version>
typedef enum hipsolverDnParams_t {};
#endif 
```

Reply:
I removed it since in the build log it showed the variable is declared in rocsolver and a redeclaration. I have updated this branch where I moved hipsolverDnParams_t under HIP_VERSION < 50631061
It passes all checks.

Reply:
@shbiswas834 This will fail on versions between 5.0 - 6.1 with error saying not declared. 
Can we wrap this one seperately and don't move below 5.0. 
```
#if HIP_VERSION < <6.2 version number>
typedef enum hipsolverDnParams_t {};
#endif 

....
#if HIP_VERSION < 50631061
typedef hipsolverHandle_t hipsolverDnHandle_t;
typedef void* hipsolverGesvdjInfo_t;
typedef void* hipsolverSyevjInfo_t;
```